### ISSUE TITLE
jobdb.db is not being written to.

### ISSUE THREAD

**Issue by minorsecond** [minorsecond/Timeclock#36]
This db is used to store the actual timesheet (amount of hours worked for each project, project name, abbrev, and lead name). The code doesn't yet write data to it.

Data should be written to it when user clocks out for lunch, switches tasks, takes a break, or goes home.

**Comment by minorsecond** [comment]
Variables are being filled with data for db fields.

**Comment by minorsecond** [comment]
Latest push fixes this. Needs testing.

**Comment by NotTheEconomist** [comment]
I'm working on a patch that will move the data management over to SQLAlchemy, which uses a more declarative style of table definition. Check out my `sqlalchemy-convert` branch to see what I'm talking about -- still in-progress.

**Comment by minorsecond** [comment]
@NotTheEconomist Should I stop working on sqlite? I've not used SQLAlchemy so I'm not sure how it would integrate with what I've got.

**Comment by NotTheEconomist** [comment]
@minorsecond No, SQLAlchemy is just a wrapper around sqlite. Most of it will transfer directly. Mostly I need to know what you want to DO with this data! I'd love it if you could focus on that, either in code or in documentation.

**Comment by minorsecond** [comment]
@NotTheEconomist  The idea is to have a database sticking around for persistent storage, so that it'll be possible to recall date for the past x months/years/whatever, and do other things with it (not sure what yet but want to leave that open). 

There are two tables currently, timesheet and jobsdb. Timesheet stores all of the data, all clock-ins and clock-outs, breaks, now tasks, etc. Jobsdb is the actual timesheet (might be good to switch the names around, come to think of it). This will store the time worked for each task, each day.

I can add more documentation in the code. I probably should at least explain how the data is pulled from the tables and manipulated. I think the DB portion is pretty close to being complete (knock on wood). I need to look at the connection code at the beginning of the script as I'm not entirely sure I need two separate connections for two different tables.

**Comment by NotTheEconomist** [comment]
What I had envisioned and partially implemented is one table with information about each employee, one table with information about each job to track, and one table with information with clock-ins and clock-outs that references the employee and job. This actually eliminates the need to track breaks as a separate entity, since any time "clocked out" is equal

**Comment by NotTheEconomist** [comment]
Check out https://trello.com/b/DyPR0k2E/planning for what I was thinking of.

**Comment by minorsecond** [comment]
I see. It looks like we are shooting at two different things here. My idea was to build an employee-side project management tool, to assist in keeping track of which projects were worked on, and how many hours were worked on each.

**Comment by NotTheEconomist** [comment]
That's what I'm envisioning, too. However: this is your project. Tell me what you're interested in and I'll help if I can. If you have a Trello account I can invite you to that org so you can organize it as you see fit :)

**Comment by minorsecond** [comment]
The only thing I'm confused about is the "information about each employee." For an employee-side tool, would it be necessary to track multiple employees? I like what you said about breaks. It hadn't occurred to me that it's unnecessary - we can just change "lunch" to "stop working" or something, and have the user log back in when returning. 

I just created a Trello account.  username is robertwardrup.

**Comment by NotTheEconomist** [comment]
So you're imagining a separate application for each employee, or for this to be used for single-employee companies (e.g. I.T. contracting, etc)? We could start by querying the `employees` table, and prompting to create a user if there is none available. If it finds a row, it uses that employee for the app session. That means in most use cases the `employees` table has one entry, but it still allows a single person to use the same application for different identities (e.g. I track my jobs for CompanyA with adsmith@companyA.com and my jobs for CompanyB with adsmith@companyB.com)

**Comment by minorsecond** [comment]
The company at which I work is "project-based," in that I charge hours depending on which project I'm working on. I've I'm working on Project A, task B, I use abbrev 11111. If I work on project A, task C, I use 22222. Different charge numbers for different tasks. At the end of the day, we report hours for each abbrev that we worked under. We occasionally are audited, at which time we could be asked to present our copy of the timesheet.

So I had envisioned each person running this (whoever wants to use it). But, after that explanation, I see the value in your solution. It's scalable in your case. Let's do it.
